packages/dds/tree/src/test/feature-libraries/modular-schema/modularChangeFamily.spec.ts

```diff
@@ -1048,6 +1064,195 @@ describe("ModularChangeFamily", () => {
 		});
 	});
 
+	describe("adds missing builds", () => {
+		const aMajor = mintRevisionTag();
+		const a1 = { major: aMajor, minor: 1 };
+		const a2 = { major: aMajor, minor: 2 };
+		const bMajor = mintRevisionTag();
+		const b1 = { major: bMajor, minor: 1 };
+		const cMajor = mintRevisionTag();
+		const c1 = { major: cMajor, minor: 1 };
+		const u1 = { minor: 1 };
+
+		const node2 = singleJsonCursor(2);
+		const node2Chunk = treeChunkFromCursor(node2);
+		const node3 = singleJsonCursor(3);
+		const node3Chunk = treeChunkFromCursor(node3);
+
+		const nodesArray: [DeltaDetachedNodeId, TreeChunk][] = [
+			[a1, node1Chunk],
+			[a2, node2Chunk],
+			[b1, node3Chunk],
+		];
+		const nodeMap = nestedMapFromFlatList(
+			nodesArray.map(([{ major, minor }, chunk]) => [major, minor, chunk]),
+		);
+
+		const getDetachedNode = ({ major, minor }: DeltaDetachedNodeId) => {
+			return tryGetFromNestedMap(nodeMap, major, minor);
+		};
+
+		it("preserves relevant builds that are present in the input", () => {
+			const input: ModularChangeset = {
+				fieldChanges: new Map([]),
+				builds: new Map([
+					[undefined, new Map([[brand(1), node2Chunk]])],
+					[aMajor, new Map([[brand(2), node2Chunk]])],
```

Comment:
Not having a1 in here means that this test is also testing that missing relevant builds are added.
This muddies the purpose of the test. For example, it might lead the test to fail even if the implementation does preserve relevant builds that are present in the input. You can't always fully isolate all aspects of the contract you're testing, but that's one aspec we can keep out from this test.

You could either add a1 here, or remove a1 from the relevant list (and the expected output). The advantage of adding it in here is that it means you test a case where there's more than one entry in an inner map.